src/server/components/dashboard/listsItems/helpers.ts

```diff
@@ -1,104 +1,37 @@
 import { logger } from "server/services/logger";
-import { prisma } from "server/models/db/prisma-client";
-import { userIsListPublisher } from "server/components/dashboard/helpers";
 
-import type { NextFunction, Request, Response } from "express";
-import type { List } from "server/models/types";
-import { ListItemGetObject } from "server/models/types";
+import type { NextFunction, Request } from "express";
 import { HttpException } from "server/middlewares/error-handlers";
-import { ListItemConfirmationPages, ListItemUrls } from "server/components/dashboard/listsItems/types";
-import { ListItemJsonData } from "server/models/listItem/providers/deserialisers/types";
-import { dashboardRoutes } from "server/components/dashboard";
+import {prisma} from "server/models/db/prisma-client";
+import {ListItemRes} from "server/components/dashboard/listsItems/types";
 
-export async function redirectIfUnauthorised(req: Request, res: Response, next: NextFunction): Promise<void> {
+export async function redirectIfUnauthorised(req: Request, res: ListItemRes, next: NextFunction): Promise<void> {
   try {
-    const { listId } = req.params;
-
-    if (!Number.isInteger(Number(listId))) throw new Error("listId is not a number");
-
-    const listData = (await prisma.list.findUnique({
-      where: {
-        id: Number(listId),
-      },
-    })) as List;
-
-    const userCanPublishList = userIsListPublisher(req, listData);
+    const { list } = res.locals;
+    const userCanPublishList = await req.user?.isListPublisher(list!.id) ?? false;
 
     if (!userCanPublishList) {
-      const err = new HttpException(403, "403", "User is not authorized to access this list.");
+      const err = new HttpException(403, "403", "User is not authorised to access this list.");
       return next(err);
     }
 
     next();
   } catch (error) {
     logger.error(`redirectIfUnauthorised Error: ${(error as Error).message}`);
-    const err = new HttpException(403, "403", "Unable to validate this request.  Please try again.");
+    const err = new HttpException(403, "403", "Unable to validate this request. Please try again.");
     return next(err);
   }
 }
 
-export function getConfirmationPages(listItemUrls: ListItemUrls): ListItemConfirmationPages {
-  return {
-    publish: {
-      path: "dashboard/list-item-confirm-publish",
-      postActionPageUrl: listItemUrls.listItemPublish,
-    },
-    unpublish: {
-      path: "dashboard/list-item-confirm-unpublish",
-      postActionPageUrl: listItemUrls.listItemPublish,
-    },
-    requestChanges: {
-      path: "dashboard/list-item-confirm-changes",
-      postActionPageUrl: listItemUrls.listItemRequestChanges,
-    },
-    updateLive: {
-      path: "dashboard/list-item-confirm-update",
-      postActionPageUrl: listItemUrls.listItemUpdate,
-    },
-    updateNew: {
```

Comment:
@alisalaman-cyb - what's updateNew for?

Reply:
this was intended for that weird condition for the radio button label.  

updateNew - radio button shows "Publish"
 - new provider details submitted
 - request changes
 - provider submits edits

updateLive - radio button shows "Update live version"
 - provider details submitted and then published by post
 - post need to request changes after initially being published (future behaviour will not unpublish the list item, it'll stay live / published)
 - provider submits edits

Reply:
oke doke. Smokies are passing.. I can see that they're tested for.. theoretically have un-broken things I broke..